lib/internal/Magento/Framework/Model/ResourceModel/Db/AbstractDb.php

```diff
@@ -367,13 +367,48 @@ public function load(\Magento\Framework\Model\AbstractModel $object, $value, $fi
         return $this;
     }
 
+    /**
+     * Load an object for update (locks record)
+     *
+     * @param \Magento\Framework\Model\AbstractModel $object
+     * @param mixed $value
+     * @param string $field field to load by (defaults to model id)
+     * @return $this
+     */
+    public function loadForUpdate(\Magento\Framework\Model\AbstractModel $object, $value, $field = null)
```

Comment:
Could you please slightly refactor `load` and `loadForUpdate` to avoid code duplication between them.

Reply:
Moved some of the function calls that are duplicated on both methods to a new method `_beforeObjectLoad`. The `beforeLoad` and `_beforeLoad` method names are already in use.